recipes/gperf/all/conanfile.py

```diff
@@ -56,6 +58,10 @@ def _configure_autotools(self):
 
     def _build_configure(self):
         with tools.chdir(self._source_subfolder):
+            try:
+                self.run("autogen.sh", run_environment=True)
+            except ConanException:
```

Comment:
Please get rid of the `try/except`.
This hides problems, as there is currently one on Windows.

Reply:
Also, the `wget` inside `autogen.sh` is unacceptable.

Reply:
@madebr I am waiting on the answer about my question on using autoconf (https://github.com/conan-io/conan-center-index/pull/6226#issuecomment-882116709) before patching autogen.sh since if we can use that there is no need to call autogen.sh.

Reply:
`autoconf.sh` has the following content: https://git.savannah.gnu.org/gitweb/?p=gperf.git;a=blob;f=autogen.sh;h=36437586720dabd89ced01b9d4f12e64b01b5cc2;hb=HEAD

It needs gnulib, which is available in conan-center-index (gnulib/20200224)
So you need to add `gnulib` as a build requirement (in addition to autoconf/2.71).
And set the `GNULIB_TOOL` environment variable to the gnulib executable found in the cci gnulib package.

That way, no internet connection is needd while running `autogen.sh`.

Reply:
@madebr Do you mean autogen.sh? (you typed autoconf.sh) My question is. If we run autoconf, does it still invoke autogen.sh? Or is autogen.sh not needed in that case?

Reply:
`autogen.sh` runs `libtool` and `make -f Makefile.devel`.
`Makefile.devel` runs  `autoconf`.

Reply:
We want to skip their utility script and go straight to building

Reply:
The `autogen.sh` script does more then `autoreconf`.
But perhaps running `autoreconf -fiv` is enough.